src/Microsoft.VisualStudio.Editors/MyApplication/MyApplicationProperties.vb

```diff
@@ -1397,6 +1397,20 @@ Namespace Microsoft.VisualStudio.Editors.MyApplication
                     Return MyApplicationProperties.ApplicationTypeFromOutputType(
                                 CUInt(props3.OutputType),
                                 props3.MyType) = ApplicationTypes.WindowsApp
+                Else
+                    ' CPS projects return DynamicTypeBrowseObject for its BrowseObject, which doesn't implement VBProjectProperties3 (or anything else) so we just try to get these properties by ICustomTypeDescriptor
```

Comment:
This is a roundabout way of reading `DTE.Project.Properties`.

Reply:
Yes, but its also a static method that only has the IVSHierarchy to act on. I can change the callers to pass in the DTE Project, or get the VSHPROPID_ExtObject property and cast it to `EnvDTE.Project` if you think it would be better, perhaps for reasons I'm not aware of, but based purely on code complexity it comes out about the same in my eyes.

Reply:
Folks in the future are going to read this code and going to try and understand why its reading the properties via this very obscure way that only works for CPS, there is no guaranteed handshake between CPS and consumers that it implements ICustomTypeDescriptor, only that its implements IDispatch.

Reply:
I would even consider just always using Project.Properties and remove the CPS-only/legacy-only paths.

Reply:
I guess that makes sense, I essentially did that with the `MyApplication` property anyway.

Reply:
Found some helper methods, the new code is much neater. Thank you :D